core/sql/parser/sqlparser.y

```diff
@@ -14684,6 +14759,11 @@ optional_limit_spec : TOK_LIMIT NUMERIC_LITERAL_EXACT_NO_SCALE
 
 dml_statement : dml_query { $$ = $1; }
 
+               | with_clause_list dml_query
```

Comment:
hi, Hans, Dave,
the current change doesn't get higher number of conflicts, it is still 73/12, but if I tried to move the rule into query_expression, it will increase the shift/reduce conflicts, I still cannot find a good way to write a correct syntax.  It is rather difficult to find the root cause of shift/reduce conflict